cli: TestUnavailableZip failed

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13023260?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13023260?buildTab=artifacts#/) on master @ [dd062fdfb54f3d44c9b82c8e51d7ed767a46f124](https://github.com/cockroachdb/cockroach/commits/dd062fdfb54f3d44c9b82c8e51d7ed767a46f124):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip4280370485
    zip_test.go:497: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip4280370485
--- FAIL: TestUnavailableZip (72.80s)
=== RUN   TestUnavailableZip/server_1
[debug --host=127.0.0.1:41177 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
[debug --host=127.5.142.1:44651 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    zip_test.go:444: panic: runtime error: invalid memory address or nil pointer dereference
    --- FAIL: TestUnavailableZip/server_1 (32.08s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-inf-prs @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34196

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13030481?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13030481?buildTab=artifacts#/) on master @ [af1fda504a6d6eb040aea4afc1323866e8695bc5](https://github.com/cockroachdb/cockroach/commits/af1fda504a6d6eb040aea4afc1323866e8695bc5):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip2610299639
    zip_test.go:497: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip2610299639
--- FAIL: TestUnavailableZip (75.26s)
=== RUN   TestUnavailableZip/server_2
    zip_test.go:480: panic: runtime error: invalid memory address or nil pointer dereference
    --- FAIL: TestUnavailableZip/server_2 (38.19s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13033044?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13033044?buildTab=artifacts#/) on master @ [eb4a9168389f341a573ac91b75e717ada0e3a04a](https://github.com/cockroachdb/cockroach/commits/eb4a9168389f341a573ac91b75e717ada0e3a04a):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip3366119228
    zip_test.go:497: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip3366119228
--- FAIL: TestUnavailableZip (74.52s)
=== RUN   TestUnavailableZip/server_1
[debug --host=127.0.0.1:40945 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    zip_test.go:444: panic: runtime error: invalid memory address or nil pointer dereference
    --- FAIL: TestUnavailableZip/server_1 (32.19s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13034949?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13034949?buildTab=artifacts#/) on master @ [ac26e8785b29f0dd53250d96de9ab1a2c2a9cebf](https://github.com/cockroachdb/cockroach/commits/ac26e8785b29f0dd53250d96de9ab1a2c2a9cebf):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip3355382872
    zip_test.go:497: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip3355382872
--- FAIL: TestUnavailableZip (75.79s)
=== RUN   TestUnavailableZip/server_2
[debug --host=127.137.99.95:43947 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    zip_test.go:480: panic: runtime error: invalid memory address or nil pointer dereference
    --- FAIL: TestUnavailableZip/server_2 (38.41s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -441,7 +441,7 @@ func TestUnavailableZip(t *testing.T) {
 
 		out, err := c.RunWithCapture(debugZipCommand)
 		if err != nil {
-			t.Fatal(err)
+			c.fail(err)
 		}
 
 		// Assert debug zip output for cluster, node 1, node 2, node 3.
@@ -477,7 +477,7 @@ func TestUnavailableZip(t *testing.T) {
 
 		out, err := c.RunWithCapture(debugZipCommand)
 		if err != nil {
-			t.Fatal(err)
+			c.fail(err)
 		}
 
 		// Assert debug zip output for cluster, node 2.

